Core/HLE/sceKernelModule.cpp

```diff
@@ -557,17 +557,17 @@ void __KernelModuleShutdown()
 // Sometimes there are multiple LO16's or HI16's per pair, even though the ABI says nothing of this.
 // For multiple LO16's, we need the original (unrelocated) instruction data of the HI16.
 // For multiple HI16's, we just need to set each one.
-struct HI16RelocInfo
-{
+struct HI16RelocInfo {
 	u32 addr;
 	u32 data;
 };
-static void WriteVarSymbol(u32 exportAddress, u32 relocAddress, u8 type, bool reverse = false)
-{
+static void WriteVarSymbol(u32 exportAddress, u32 relocAddress, u8 type, bool reverse = false) {
 	// We have to post-process the HI16 part, since it might be +1 or not depending on the LO16 value.
 	static u32 lastHI16ExportAddress = 0;
+	// This is just a list of recent HI16s for the same export address.
 	static std::vector<HI16RelocInfo> lastHI16Relocs;
 	static bool lastHI16Processed = true;
+	static bool lastHI16Reverse = false;
```

Comment:
I'm wondering if these shouldn't belong to the module instead of being static (effectively global)? Or maybe it doesn't matter?

Reply:
It can't be interleaved.  These values shouldn't even last for the entire lifetime of the module.

Supposing a module importing 20 variables, each using the common case of a single HI16/LO16 pair.  In that case, the `lastHI16Relocs` vector would never hold more than one item, and would be cleared 20 times on link within  `if (lastHI16ExportAddress != exportAddress || lastHI16Reverse != reverse) {`.  It would be cleared another 20 times on unlink.

It's not some long lived list of relocations.  It's just for the HI16s that need to be adjusted for the next LO16s.

-[Unknown]

Reply:
I moved it to a state object - I suppose if two modules both imported exactly the same variable there might be a similar problem?  But hopefully this makes it clearer to everyone that this isn't some module state.

-[Unknown]

Reply:
That's much better! I avoid writing code with statics like that these days, prefer to let the type/lifetime system of the language enforce proper usage...